### PR TITLE
Fixed #26192 -- Added control for the value expression is in both order_by  and select clause in the same query.

### PR DESCRIPTION
**Pull request by cansarigol**
Ticket: [#26192](https://code.djangoproject.com/ticket/26192)

### CODE REVIEW

File: tests/ordering/tests.py [hunk 1]
```diff
@@ -317,6 +320,20 @@ def test_order_by_fk_attname(self):
             attrgetter("headline")
         )
 
+    def test_order_by_value_expression_with_annotate(self):
+        msg = 'Cannot resolve expression type, unknown output_field'
+        with self.assertRaisesMessage(FieldError, msg):
+            list(Article.objects.annotate(headline2=Value('headline')))
+        self.assertQuerysetEqual(
+            Article.objects.annotate(headline2=Value('headline', output_field=CharField())), [
```
Comment (MahdiZareie) [inline tests/ordering/tests.py]: Didn't you forget to add an order_by here?

Comment (cansarigol) [reply tests/ordering/tests.py]: I forgot it thanks 👍

**Comment by felixxm** [comment]
@cansarigol This is not a correct approach. We cannot add a restriction for querysets that are valid in all databases except PostreSQL. Moreover expressions from `ORDER BY` clause don't need to be included in `SELECT` clause.  It seems that you want to add this restriction only because in other place we check `output_type` of selected columns, this is confusing.

IMO removing constants from `ORDER BY` clause is the only feasible solution, e.g. (only draft)
```diff
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1843,6 +1843,8 @@ class Query(BaseExpression):
                 )
         if errors:
             raise FieldError('Invalid order_by arguments: %s' % errors)
+        # Remove constants from ordering.
+        ordering = tuple([item for item in ordering if not isinstance(self.annotations.get(item), Value)])
         if ordering:
             self.order_by += ordering
         else:
```

**Comment by cansarigol** [comment]
Hi @felixxm, sorry about improper solution. I just wanted to explain to the user that if you use Value expression column into order, order of your query result won't be what you expect. (Maybe a warning would be added instead of restriction). 
I read the removing solution from ticket comments, it didn't appropriate for me.

**Pull request closed without merging**